lib/player.rb

```diff
@@ -0,0 +1,87 @@
+require_relative '../lib/tilebag'
+
+module Scrabble
+  class Player
+    attr_reader :name, :plays, :tiles
+    def initialize(name, debug = false)
+      @name = name
+      @plays = []
+      @score_array = []
+      @tiles = []
+      @debug = debug
+    end#initialize
+
+    def play(word)
+      if won?
+        false
+      else
+        @plays << word.upcase
+        delete_tiles_from_tiles_array(word)
+        word_score = Scrabble::Scoring.score(word)
+        @score_array << word_score
+        return word_score
+      end
+    end #play word end
+
+    def delete_tiles_from_tiles_array(word)
+      word = word.upcase
+      word.split("").each do |letter|
+        if @tiles.include?(letter)
+          @tiles.slice!(@tiles.index(letter))
+        else
+          raise ArgumentError.new("You cannot play this word.")
+        end
+      end
+    end
+
+    def total_score
+      return @score_array.inject(0){|sum, x| sum + x}
+    end #total score end
+
+    def won?
+      if total_score > 100
+        return true
+      else
+        return false
+      end
+    end #won end
+
+    def highest_scoring_word_score
+      if @score_array  == []
+        raise ArgumentError.new("No words to compare")
+      else
+        index = @score_array.index(@score_array.max)
+        max_word = @plays[index]
+        max_score = @score_array.max
+        return max_word, max_score
```

Comment:
This method could be simpler if we don't keep separate `@plays` and `@score_array` arrays. Instead we could just have `@plays` and use the functionality from the `Scoring` class to determine the scores as-needed:

```ruby
def highest_scoring_word_score
  raise ArgumentError.new("No words to compare") if @plays.empty?

  best_word = Scoring.highscore_from(@plays)
  return best_word, Scoring.score(best_word)
end
```

Reply:
In addition, the project requirements specify two methods, `highest_scoring_word` and `highest_word_score`. It looks like y'all have combined the functionality of those two methods into this method `highest_scoring_word_score`.

In the future we should strive to match the requirements specification as much as possible, so this method should be broken up into the two separate methods that each calculate one part of the two return values that this method is currently returning.